### PR TITLE
Adding tag rc and release commands to k3s_release

### PR DESCRIPTION
**Pull request by galal-hussein**
- Add `tag-rc-release` command to cut an rc according to the config file passed to the k3s_release command
- Add `tag-release` command to cut a **"draft"** release associated with the release notes.

### CODE REVIEW

File: cmd/k3s_release/tag_rc_release.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	if err != nil {
+		logrus.Fatalf("failed to read config file: %v", err)
+	}
+
+	client, err := k3s.NewGithubClient(ctx, release.Token)
+	if err != nil {
+		logrus.Fatalf("failed to initialize a new github client from token: %v", err)
+	}
+
+	err = release.CreateRelease(ctx, client, true)
```
Comment (briandowns) [inline cmd/k3s_release/tag_rc_release.go:35]:
```suggestion
	return release.CreateRelease(ctx, client, true)
```

File: cmd/k3s_release/tag_release.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	if err != nil {
+		logrus.Fatalf("failed to read config file: %v", err)
+	}
+
+	client, err := k3s.NewGithubClient(ctx, release.Token)
+	if err != nil {
+		logrus.Fatalf("failed to initialize a new github client from token: %v", err)
+	}
+
+	err = release.CreateRelease(ctx, client, false)
```
Comment (briandowns) [inline cmd/k3s_release/tag_release.go:35]:
```suggestion
	return release.CreateRelease(ctx, client, false)
```

File: release/release.go [hunk 3]
```diff
@@ -472,6 +473,30 @@ func findInURL(url, regex, str string) []string {
 	return submatch
 }
 
+// LatestRC will get the latest rc created for the k8s version in either rke2 or k3s
+func LatestRC(ctx context.Context, repo string, k8sVersion string, client *github.Client) (string, error) {
```
Comment (briandowns) [inline release/release.go:477]:
```suggestion
func LatestRC(ctx context.Context, repo, k8sVersion string, client *github.Client) (string, error) {
```

File: release/k3s/k3s.go [hunk 4]
```diff
@@ -638,3 +640,44 @@ func cleanGitRepo(dir string) error {
...
+			latestRc, err := release.LatestRC(ctx, k3sRepo, r.NewK8SVersion, client)
+			buff, err := release.GenReleaseNotes(ctx, k3sRepo, latestRc, oldName, client)
+			if err != nil {
+				return err
+			}
+			opts.ReleaseNotes = buff.String()
+		}
+		_, err := repository.CreateRelease(ctx, client, opts)
+		if err != nil {
+			githubErr := err.(*github.ErrorResponse)
```
Comment (briandowns) [inline release/k3s/k3s.go:670]: Can we use errors.Is or errors.As there instead of the cast?

**Comment by briandowns** [comment]
I wonder if we can have a creatRelease function be called from both sub commands and that function calls the release package function, and eliminate some of the repetition.

File: release/k3s/k3s.go (hunk 4 above)
Comment (galal-hussein) [reply release/k3s/k3s.go:670]: I checked github library that we use, and unfortunately I cant find good documentation for the error code, it seemed to be unstructured, and I found that comment in the code:
```
GitHub error responses structure are often undocumented and inconsistent.
Sometimes error is just a simple string (Issue #540).
In such cases, Message represents an error message as a workaround.
```
here is the referenced issue https://github.com/google/go-github/issues/540

**Comment by galal-hussein** [comment]
> der if we can have a creatRelease function be called from both sub commands and that function calls the release package function, and eliminate some of the repetition.

yeah thats currently what happens actually:

- tag-rc-release:
```
release.CreateRelease(ctx, client, true) --> repository.CreateRelease()
```
- tag-release:
```
release.CreateRelease(ctx, client, false) --> repository.CreateRelease()
```

**Review by briandowns — APPROVED** [review]

**Pull request merged**